pkg/util/validation/validate.go

```diff
@@ -61,6 +61,29 @@ const (
 	TooManyHAClusters = "too_many_ha_clusters"
 )
 
+// allDiscardedSampleReasons is used when cleaning up metrics for a user.
+var allDiscardedSampleReasons = []string{
+	missingMetricName,
+	invalidMetricName,
+	greaterThanMaxSampleAge,
+	maxLabelNamesPerSeries,
+	tooFarInFuture,
+	invalidLabel,
+	labelNameTooLong,
+	duplicateLabelNames,
+	labelsNotSorted,
+	labelValueTooLong,
+	RateLimited,
+	TooManyHAClusters,
+}
+
+var allDiscardedMetadataReasons = []string{
```

Comment:
In the distributor could also be RateLimited, perUserMetadataLimit, perMetricMetadataLimit.